packages/token-sdk/src/token-fetcher.ts

```diff
@@ -57,29 +67,45 @@ export class TokenFetcher {
         }
       }
     }
-    return this._cache[mintString];
+    return { ...this._cache[mintString] };
   }
 
   public async findMany(addresses: Address[]): Promise<ReadonlyTokenMap> {
     const mints = AddressUtil.toPubKeys(addresses);
-    const misses = mints.filter((mint) => !this._cache[mint.toBase58()]);
+    const misses = mints.filter(
+      (mint) =>
+        !this._cache[mint.toBase58()] || MetadataUtil.isPartial(this._cache[mint.toBase58()])
+    );
 
     if (misses.length > 0) {
+      console.log(`Fetching mint info for ${misses.length} mints...`);
       const mintInfos = (
-        await getMultipleParsedAccounts(this.connection, misses, ParsableMintInfo)
+        await pTimeout(
+          getMultipleParsedAccounts(this.connection, misses, ParsableMintInfo),
+          this.timeoutMs
+        )
       ).filter((mintInfo): mintInfo is MintInfo => mintInfo !== null);
       invariant(misses.length === mintInfos.length, "At least one mint info not found");
       misses.forEach((mint, index) => {
         const mintString = mint.toBase58();
         this._cache[mintString] = {
-          mint,
+          mint: mintString,
           decimals: mintInfos[index].decimals,
         };
       });
 
       let next = misses;
       for (const provider of this.providers) {
-        const metadatas = await provider.findMany(next);
+        let metadatas: Record<string, Partial<TokenMetadata> | null>;
+        try {
+          console.log(
+            `Fetching metadata for ${next.length} mints from ${provider.constructor.name}...`
```

Comment:
IIRC, console.log in the normal path is not often seen in the SDK.
How about moving it the catch clause?

Reply:
In my testing it's showing up in the GitHub Actions log when used - specifically under the `Generate Tokens` step here:
https://github.com/orca-so/orca-mintlists-demo/actions/runs/4321528988/jobs/7542865867